mitiq/executor/executor.py

```diff
@@ -151,6 +151,15 @@ def evaluate(
                 for circuit_with_measurements in observable.measure_in(circuit)
             ]
             result_step = observable.ngroups
+        elif (
+            observable is not None
+            and self._executor_return_type not in MeasurementResultLike
```

Comment:
```suggestion
            and self._executor_return_type not in MeasurementResultLike
            and self._executor_return_type not in DensityMatrixLike
```

The observable is compatible with DensityMatrixLike return types too. This may fix most of the problems.